### ISSUE TITLE
No turn issued for right turn on un-named roads

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#4414]
The following route should include a `turn right` instruction:

![screen shot 2017-08-15 at 3 33 49 pm](https://user-images.githubusercontent.com/1892250/29339457-2aeeecb0-81cf-11e7-8e17-b44b3a7cc4e6.png)

Here is satellite imagery of the intersection:
![screen shot 2017-08-15 at 3 35 39 pm](https://user-images.githubusercontent.com/1892250/29339544-7492a352-81cf-11e7-9518-924823de8245.png)

And a link to the route (note that I've mapped the missing slip-road, so it may no longer generate the route shown above):
http://map.project-osrm.org/?z=18&center=52.156971%2C9.874869&loc=52.156038%2C9.872396&loc=52.157559%2C9.875057&hl=en&alt=0

One problem seems to be a lack of tagging - none of these roads have a `name` tag, and there's a missing sliproad.  It's also debatable whether the road heading east is `secondary` or should instead be something smaller.

However, the northbound road does have a `ref` tag, and the east-bound road does not.  I feel this should be enough to trigger a turn instruction - currently, our obvious turn detection only seems to be looking at road priorities and names.  I feel that *despite* the lack of tags, there should be enough information here to detect a proper turn.

Here's a test case that currently fails but I feel should pass:

```
@routing @car
Feature: Car - Handle driving

    Background:
        Given the profile "car"
        And a grid size of 10 meters

    Scenario: No turn instruction when turning from unnamed onto unnamed
        Given the node map
            """
             a
            |
            |
            |
            |
            b----------------c
            |
            |
            |
            |
            |
            |
            d
            """

        And the ways
            | nodes | highway     | name | ref   |
            | ab    | trunk_link  |      |       |
            | db    | secondary   |      | L 460 |
            | bc    | secondary   |      |       |

        When I route I should get
            | from | to | route    | turns                    |
            | d    | c  | db,bc,bc | depart,turn right,arrive |
```

**Comment by danpat** [comment]
For completeness, also note that reversing the coordinates in the example above leads to:

![screen shot 2017-08-16 at 8 42 11 am](https://user-images.githubusercontent.com/1892250/29372014-d3733078-825e-11e7-9dd3-65bf713bae25.png)

This isn't so easy to fix with the addition of the slip road, and highlights the underlying bug of a shortage of information in OSM.

**Comment by danpat** [comment]
@oxidase do you know if another test case covers this scenario?  If so, we can close this.

**Comment by oxidase** [comment]
Fixed by d166fc6 and the test is committed in #5058
